lib/IRGen/IRGenModule.cpp

```diff
@@ -781,18 +781,37 @@ static StringRef encodeForceLoadSymbolName(llvm::SmallVectorImpl<char> &buf,
   return os.str();
 }
 
+class TargetLinkerOptionGenerator {
+  static std::string Quoted(StringRef Name) {
+    if (Name.find(' ') == StringRef::npos)
+      return Name.str();
+    return "\"" + Name.str() + "\"";
+  }
+
+  const llvm::Triple T;
+
+public:
+  TargetLinkerOptionGenerator(const std::string &Triple) : T(Triple) {}
+  std::string encode(StringRef Library) const {
+    if (T.isWindowsMSVCEnvironment() || T.isWindowsItaniumEnvironment())
+      return "/DEFAULTLIB:" +
```

Comment:
Nitpick: can you use a SmallString here? I know there won't be too many of these options but still.